### PR TITLE
Validate whisper beginning_timestamp_token_id bounds

### PR DESCRIPTION
**Pull request by apsonawane**
This pull request improves validation for the Whisper model's `beginning_timestamp_token_id` parameter in both runtime and test code. The main focus is to ensure that invalid values for this parameter are caught early, preventing out-of-range errors during inference and processing.

**Validation improvements for Whisper timestamp token ID:**

* Added a check in `BeamSearchParameters::SetSubgraphParameters` to enforce that `beginning_timestamp_token_id` is within the valid range `[0, vocab_size)` for Whisper models, throwing an error if not.
* Added a similar runtime check in `TimestampLogitsProcessor` to ensure `beginning_timestamp_token_id_` is within bounds before processing logits.

**Unit test enhancements:**

* Introduced new tests in `beam_search_test.cc` to verify that `SetSubgraphParameters` correctly rejects negative or out-of-range `beginning_timestamp_token_id` values and accepts valid ones.

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/transformers/beam_search_parameters.cc [hunk 1]
```diff
@@ -152,6 +152,16 @@ void BeamSearchParameters::SetSubgraphParameters(int vocabulary_size, int heads,
     ORT_THROW("vocab_size attribute (", vocab_size,
               ") cannot exceed decoder subgraph logits width (", vocabulary_size, ")");
   }
+
+  // Whisper timestamp logits processing indexes token-score spans by
+  // beginning_timestamp_token_id. Reject out-of-range ids up front.
+  if (model_type == IGenerationParameters::kModelTypeWhisper &&
+      logits_processor == IGenerationParameters::kLogitsProcessorTypeWhisper) {
```
Comment (tianleiwu) [inline onnxruntime/contrib_ops/cpu/transformers/beam_search_parameters.cc:164]: `SetSubgraphParameters()` runs during `SetupSubgraphExecutionInfo()`, before any `BeamSearchBase` is constructed and calls `ParseFromInputs()`. At this point `IGenerationParameters::logits_processor` has no default initializer; it is assigned only from input 11 in `ParseFromInputs()`. This condition therefore reads an indeterminate value, so it can either skip the intended validation or spuriously reject model initialization. The new tests mask the lifecycle issue by assigning `parameters.logits_processor` manually before calling this method. Please move this conditional validation to immediately after `ParseFromInputs()` assigns `logits_processor` (the subgraph has populated `vocab_size` by then), and keep the `TimestampLogitsProcessor::Process()` check as defense in depth. Merely defaulting the member to zero would remove the undefined read but would still skip validation when a request later enables timestamp processing.

**Review by tianleiwu — CHANGES_REQUESTED** [review]
Requesting changes for one correctness issue: the new setup-time guard reads a runtime-input field before it is initialized, so model initialization has undefined behavior and the early validation may be skipped or fire spuriously. The runtime `Process` guard is otherwise sound and should remain as defense in depth.

File: onnxruntime/test/contrib_ops/beam_search_test.cc [hunk 2]
```diff
@@ -49,6 +49,46 @@ TEST(BeamSearchParametersTest, SetSubgraphParametersUsesSubgraphSizeWhenAttribut
...
+}
+
+TEST(BeamSearchParametersTest, AcceptsValidWhisperBeginningTimestampTokenId) {
+  contrib::transformers::BeamSearchParameters parameters;
+  parameters.vocab_size = 128;
+  parameters.model_type = contrib::transformers::IGenerationParameters::kModelTypeWhisper;
+  parameters.logits_processor = contrib::transformers::IGenerationParameters::kLogitsProcessorTypeWhisper;
+  parameters.beginning_timestamp_token_id = 1;
+
+  EXPECT_NO_THROW(parameters.ValidateWhisperTimestampTokenId());
```
Comment (tianleiwu) [inline onnxruntime/test/contrib_ops/beam_search_test.cc:89]: Non-blocking: since the validator uses a strict upper bound, consider also testing `beginning_timestamp_token_id = vocab_size - 1`. The current valid case locks in the lower endpoint (`1`) and the rejection case covers `vocab_size`; adding the highest valid value would document both accepted ends of the interval.

**Review by tianleiwu — APPROVED** [review]
The initialization-order issue from my previous review is addressed: validation now runs after `logits_processor` is parsed, with `vocab_size` already established from the decoder subgraph. The matching processor guard provides defense in depth. I found no remaining correctness blockers; the inline note is a non-blocking test-coverage suggestion.
